test/test_sequencing_groups.py

```diff
@@ -136,5 +141,174 @@ async def test_auto_deprecation_of_old_sequencing_group(self):
         active_sgs = await self.sglayer.query(
             SequencingGroupFilter(sample_id=GenericFilter(sample.id))
         )
+
+        self.assertTrue(all(not sg.archived for sg in active_sgs))
         self.assertEqual(len(active_sgs), 1)
         self.assertEqual(updated_sample.sequencing_groups[0].id, active_sgs[0].id)
+
+    @run_as_sync
+    async def test_query_with_assay_metadata(self):
+        """Test searching with an assay metadata filter"""
+        sample_to_insert = get_sample_model()
+
+        # Add extra sequencing group
+        sample_to_insert.sequencing_groups.append(
+            SequencingGroupUpsertInternal(
+                type='exome',
+                technology='short-read',
+                platform='ILLUMINA',
+                meta={
+                    'meta-key': 'meta-value',
+                },
+                external_ids={},
+                assays=[
+                    AssayUpsertInternal(
+                        type='sequencing',
+                        external_ids={},
+                        meta={
+                            'sequencing_type': 'exome',
+                            'sequencing_platform': 'short-read',
+                            'sequencing_technology': 'illumina',
+                        },
+                    )
+                ],
+            )
+        )
+
+        # Create in database
+        sample = await self.slayer.upsert_sample(sample_to_insert)
+
+        # Query for genome assay metadata
+        sgs = await self.sglayer.query(
+            SequencingGroupFilter(
+                assay_meta={'sequencing_type': GenericFilter(eq='genome')}
+            )
+        )
+        self.assertTrue(len(sgs) == 1)
```

Comment:
Can we use `assertEqual` here too, it improves debugging (you can see how many are actually returned if it does fail).